src/commands/protocols/publish/sender/publish-schedule-messages-command.js

```diff
@@ -90,15 +93,15 @@ class PublishScheduleMessagesCommand extends ProtocolScheduleMessagesCommand {
 
         let validBids = 0;
 
-        nodes.forEach((node) => {
-            const askNumber = this.blockchainModuleManager.convertToWei(blockchain, node.ask);
-
-            const ask = this.blockchainModuleManager.toBigNumber(blockchain, askNumber);
+        await Promise.all(
+            nodes.map(async (node) => {
+                const ask = await this.getAsk(blockchain, node.id);
+                if (ask.lte(serviceAgreementBid)) {
+                    validBids += 1;
+                }
+            }),
+        );
 
-            if (ask.lte(serviceAgreementBid)) {
-                validBids += 1;
-            }
-        });
         if (validBids < minAckResponses) {
```

Comment:
this might be for another PR, but it might be worth considering filtering out the nodes with low bids alltogether to reduce messages etc. ?

Reply:
Makes sense as optimization. What are we considering now is to switch the order in dkg client, first to replicate on the network and after that to store on local node. So there won't be an issue of data duplication on local node if replication(publish) fails, and also to adapt INIT command, to check success rate, if we don't have enough success inits to not proceed with publish request. I would be interested to hear your opinion about that :)

Reply:
Makes total sense 👍️️️️️️